File: swift/Sources/sdnx/Types/CheckStatus.swift
import Foundation

public struct CheckStatus {
    public var path: [String]
    public var errors: [CheckError]
    public var defs: [String: Schema]

    public init(path: [String], errors: [CheckError], defs: [String: Schema] = [:]) {
        self.path = path
        self.errors = errors
        self.defs = defs
    }
}

File: dotnet/Sdnx.Core/Types/CheckStatus.cs
using System.Collections.Generic;

namespace Sdnx.Core
{
    public class CheckStatus
    {
        public List<string> Path { get; set; }
        public List<CheckError> Errors { get; set; }
        public Dictionary<string, Dictionary<string, SchemaValue>> Defs { get; set; }

        public CheckStatus()
        {
            Path = new List<string>();
            Errors = new List<CheckError>();
            Defs = new Dictionary<string, Dictionary<string, SchemaValue>>();
        }
    }
}
